src/lib/swiss.js

```diff
@@ -100,148 +106,57 @@ function getStandings(options, round, participants, matches) {
 
 function getMatchups(options, id, round, participants, matches) {
   matches = matches.filter(match => match.round < round)
-  var standings = getStandings(options, round, participants, matches)
-  standings.sort((a, b) => {
-    if (a.wins === b.wins) {
-      if (a.tiebreaker === b.tiebreaker) {
-        return b.seed - a.seed
-      } else {
-        return b.tiebreaker - a.tiebreaker
-      }
-    } else {
-      return b.wins - a.wins
-    }
-  })
-
-  var orderedParticipants = standings.map(s => {
-    return {
-      id: s.id,
-      seed: participants.filter(participant => participant.id === s.id)[0].seed
-    }
-  })
-  // Add BYE to the end of the list
-  if (orderedParticipants.length % 2 === 1) {
-    orderedParticipants.push({ id: null, seed: 0 })
-  }
-  var penalties = generatePenalties(options, orderedParticipants, matches)
-
-  // Split orderParticipants into reasonable chunks
-  var groups = []
-  var divisor = Math.ceil(orderedParticipants.length / 10)
-  var groupSize = orderedParticipants.length / divisor
-  groupSize = groupSize % 2 === 0 ? groupSize : groupSize + 1
-  for (var i = 0; i < orderedParticipants.length; i+= groupSize) {
-    groups.push(orderedParticipants.slice(i, i + groupSize))
+  var mappings = getMappings(participants, matches)
+  if(mappings.length % 2 === 1) {
+    // we simulate the bye having played against every team with a bye
+    // that way those teams will not get a bye again unless the matches are
+    // ridiculously better if they have another
+    // we also want it to bias toward giving byes to teams at the bottom
+    // of the standings
+    mappings.push({id:1337,
```

Comment:
We should use something more generic here for the ID. `'_'` maybe? `null` would be ideal but i'm guessing that doesn't work based on this?